scrapy/downloadermiddlewares/redirect.py

```diff
@@ -55,28 +55,25 @@ class RedirectMiddleware(BaseRedirectMiddleware):
 
     def process_response(self, request, response, spider):
         if (request.meta.get('dont_redirect', False) or
-               response.status in getattr(spider, 'handle_httpstatus_list', []) or
-               response.status in request.meta.get('handle_httpstatus_list', []) or
-               request.meta.get('handle_httpstatus_all', False)):
+                response.status in getattr(spider, 'handle_httpstatus_list', []) or
+                response.status in request.meta.get('handle_httpstatus_list', []) or
+                request.meta.get('handle_httpstatus_all', False)):
             return response
 
-        if request.method == 'HEAD':
-            if response.status in [301, 302, 303, 307] and 'Location' in response.headers:
-                redirected_url = urljoin(request.url, response.headers['location'])
+        location = None
+        if 'Location' in response.headers:
+            location = response.headers['location'].decode('latin1')
+
+        if location is not None and response.status in [301, 302, 303, 307]:
+            redirected_url = urljoin(request.url, location)
```

Comment:
Right, lemme add some tests for that.

Do you mean, besides decoding from latin1, maybe we should also do percent-encoding?
How could I check if we should do that?
The browser does show it percent-encoded in the log when using a latin1 header: `http://127.0.0.1:5000/a%C3%A7%C3%A3o`